velox/expression/VectorUdfTypeSystem.h

```diff
@@ -421,6 +428,82 @@ struct VectorWriter<Array<V>> {
   size_t offset_ = 0;
 };
 
+// A new temporary writer, to be used when the user want a proxy interface for
+// the output. Will eventually replace VectorWriter<Array>>.
+template <typename V>
+struct VectorWriter<ArrayProxyT<V>> {
+  using vector_t = typename TypeToFlatVector<Array<V>>::type;
+  using child_vector_t = typename TypeToFlatVector<V>::type;
+  using exec_out_t = ArrayProxy<V>;
+
+  void init(vector_t& vector) {
+    arrayVector_ = &vector;
+    childWriter_.init(static_cast<child_vector_t&>(*vector.elements().get()));
+  }
+
+  VectorWriter() {}
+
+  exec_out_t& current() {
```

Comment:
How do we guard against people storing references after calling current ?

Reply:
there is no guarantee, not the that protocol is there even before this diff,
however the user is not expected to call current and setOffset.
those are called by the vectorAdapter. 

But a similar problem we have is in the array proxy with the addItem().
since it returns a ref

auto& a = addItem()
auto& b = addItem(); 

In general its not safe to write to a after creating b.
I am planning to add some debug time checks that detect those miss uses 
but there is no solution to guard against that in the api itself.

Reply:
I understand, lets make it very clear when we write documentation that they shouldn't do that.